### PR TITLE
ci: Speed up tests in local development

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #\<issue number\>
- Closes #\<issue number\>

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

File: .github/workflows/extremes.yml [hunk 1]
```diff
@@ -23,12 +23,12 @@ jobs:
...
           enable-cache: "true"
           cache-suffix: ${{ matrix.python-version }}
-          cache-dependency-glob: "**requirements*.txt"
+          cache-dependency-glob: "pyproject.toml"
       - name: install-minimum-versions
         run: uv pip install tox virtualenv setuptools pandas==0.25.3 polars==0.20.3 numpy==1.17.5 pyarrow==11.0.0 scipy==1.5.0 scikit-learn==1.1.0 tzdata --system
       - name: install-reqs
         run: |
-          uv pip install -r requirements-dev.txt --system
+          uv pip install -e ".[dev]" --system
```
Comment (FBruzzesi) [inline .github/workflows/extremes.yml:31]: Somehow this is uninstalling pyarrow 11 and installs pyarrow v17.0.0.

I can see this happening consistently in the most recent PRs

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: ah well-spotted

we should probably have `[backends]` as well as `[dev]`

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: though i still don't see why that would cause this to install pyarrow 17 🤔

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: looking here it' still pyarrow 11 https://github.com/narwhals-dev/narwhals/actions/runs/12380396751/job/34556674164

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: ooh but not here, you're right! https://github.com/narwhals-dev/narwhals/actions/runs/12394099025/job/34596775668?pr=1613

Comment (FBruzzesi) [reply .github/workflows/extremes.yml:31]: True, but both:
- https://github.com/narwhals-dev/narwhals/actions/runs/12394099025/job/34596775668?pr=1613
- https://github.com/narwhals-dev/narwhals/actions/runs/12393431408/job/34594660478?pr=1614

end up with pyarrow 17

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: i think it's the new dask release

Comment (MarcoGorelli) [reply .github/workflows/extremes.yml:31]: found it, it's pyarrow-stubs! thanks again for spotting this, legend

**Pull request merged**